nixos/modules/services/security/tang.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.tang;
+in
+
+{
+  meta.maintainers = with maintainers; [ julienmalka camillemndn ];
+
+  options.services.tang = {
+    enable = mkEnableOption (mdDoc ''Tang Daemon'');
+
+    port = mkOption {
+      type = types.nullOr types.port;
+      default = 80;
+      description = mdDoc "Override the default port on which to listen for the HTTP REST API.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc "Whether to open ports in the firewall for the server.";
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      systemd.services."tangd@" = {
```

Comment:
It looks like the `tang` package ships with unit files. Needs a touch of patching to get rid of `/usr/bin`, but wouldn't it be better to use those rather than duplicating them here?

Reply:
I actually thought about this, also because it is easier to maintain, but is it common in nixpkgs to use pre-made unit files? Anyway we might rebase on another PR for the tang module!